framework/src/mesh/ConcentricCircleMesh.C

```diff
@@ -0,0 +1,660 @@
+      original_mesh.get_boundary_info().sideset_name(2) = "outer";
+    }
+    original_mesh.prepare_for_use(false);
+  }
+  else if (_portion == "full")
+  {
+    ReplicatedMesh & portion_one = cast_ref<ReplicatedMesh &>(getMesh());
+    ReplicatedMesh portion_two(portion_one);
+
+    // This is to rotate the mesh and also to reset boundary IDs.
+    MeshTools::Modification::rotate(portion_two, 90, 0, 0);
+
+    if (_pitch != 0.0)
+    {
+      // Portion 2: 2nd quadrant
+      MeshTools::Modification::change_boundary_id(portion_two, 1, 5);
+      MeshTools::Modification::change_boundary_id(portion_two, 2, 6);
+      MeshTools::Modification::change_boundary_id(portion_two, 3, 7);
+      MeshTools::Modification::change_boundary_id(portion_two, 4, 1);
+      MeshTools::Modification::change_boundary_id(portion_two, 5, 2);
+      MeshTools::Modification::change_boundary_id(portion_two, 6, 3);
+      MeshTools::Modification::change_boundary_id(portion_two, 7, 4);
+      // 'top_half'
+      portion_one.stitch_meshes(portion_two, 1, 3, TOLERANCE, true);
+
+      // 'bottom_half'
+      ReplicatedMesh portion_bottom(portion_one);
+      MeshTools::Modification::rotate(portion_bottom, 180, 0, 0);
+      MeshTools::Modification::change_boundary_id(portion_bottom, 1, 5);
+      MeshTools::Modification::change_boundary_id(portion_bottom, 2, 6);
+      MeshTools::Modification::change_boundary_id(portion_bottom, 3, 7);
+      MeshTools::Modification::change_boundary_id(portion_bottom, 4, 2);
+      MeshTools::Modification::change_boundary_id(portion_bottom, 5, 3);
+      MeshTools::Modification::change_boundary_id(portion_bottom, 6, 4);
+      MeshTools::Modification::change_boundary_id(portion_bottom, 7, 1);
+      // 'full'
+      portion_one.stitch_meshes(portion_bottom, 2, 4, TOLERANCE, true);
+
+      portion_one.get_boundary_info().sideset_name(1) = "left";
+      portion_one.get_boundary_info().sideset_name(2) = "bottom";
+      portion_one.get_boundary_info().sideset_name(3) = "right";
+      portion_one.get_boundary_info().sideset_name(4) = "top";
+    }
+    else
+    {
+      MeshTools::Modification::change_boundary_id(portion_two, 1, 5);
+      MeshTools::Modification::change_boundary_id(portion_two, 2, 1);
+      MeshTools::Modification::change_boundary_id(portion_two, 5, 2);
+      // 'top half'
+      portion_one.stitch_meshes(portion_two, 1, 1, TOLERANCE, true);
+      // 'bottom half'
+      ReplicatedMesh portion_bottom(portion_one);
+      MeshTools::Modification::rotate(portion_bottom, 180, 0, 0);
+      // 'full'
+      portion_one.stitch_meshes(portion_bottom, 2, 2, TOLERANCE, true);
+      MeshTools::Modification::change_boundary_id(portion_one, 3, 1);
+      portion_one.get_boundary_info().sideset_name(1) = "outer";
+    }
+    portion_one.prepare_for_use(false);
+  }
+  if (_portion != "top_half" && _portion != "bottom_half" && _portion != "right_half" &&
+      _portion != "left_half")
+    mesh.prepare_for_use(false);
```

Comment:
also why do you need to call this when you have already called it on the cast `ReplicatedMesh` objects which are the exact same objects?